### PR TITLE
Minor Relationship Retrieval & REST Adjustments

### PR DESCRIPTION
**Pull request by bauer-andreas**
# Minor Relationship Retrieval & REST Adjustments

## :recycle: Current situation
When developing by demo project, I encountered some smaller issues with the relationship retrieval and REST _links formatting process which were quick to fix.

## :bulb: Proposed solution
This PR adds a commit for every issue fixed.

### Problem that is solved
* b855359 This changes path parameter resolvers such that explicitly defined resolvers are prioritized over automatically retrieved resolver. This addresses one particular issue of point 7. of #223 
* ab88a58 Removes that relationships overwrite relationships of other "sources" if they point to the same destination. This address one part of the issue of point 9. of #223 
* 8dc85f1 Just some typos
* aa59ff0 Resolves point 10. of #223 
* 1cf7efe As the REST exporter will consider RelationshipDestinations with any Operation, those could compete in the name shadowing process. This change ensure that the outcome is not deepened on hashing.
* 11e3d77 Some refactoring providing better semantics for formatRelationships
* 70c6cd2 Changes the stance on https://github.com/Apodini/Apodini/pull/55#issuecomment-743771845 and enforces that Handlers must always declare all  `PathParameter` contained in their path as `@Parameter`. This mitigates some issues were such PathParameters would never be retrieved and thus wouldn't be resolved in relationships.

### Testing
Tests were added were needed.

### Reviewer Nudging
It is probably best to review this PR commit by commit as those are self contained. I hope the explanation above and in the commit message helps for a fast review.

### CODE REVIEW

**Review by theMomax — APPROVED** [review]
Everything looks pretty reasonable. Only thing I am not sure about is the change of 70c6cd2 as someone might have a strong stance against this behavior (I don't). Maybe give it a second thought who might have a problem with that and request their review in case you can think of someone. Otherwise, just go ahead🚀

**Comment by bauer-andreas** [comment]
> Everything looks pretty reasonable. Only thing I am not sure about is the change of [70c6cd2](https://github.com/Apodini/Apodini/commit/70c6cd20ad0f7d77d6f34ec896ea99313c33a53f) as someone might have a strong stance against this behavior (I don't). Maybe give it a second thought who might have a problem with that and request their review in case you can think of someone. Otherwise, just go ahead🚀

Thanks for the review.
I think the linked comment https://github.com/Apodini/Apodini/pull/55#issuecomment-743771845 describes the implications well. It really is an ill described Webservice. We previously just aimed for ignoring such definitions, as the user would need to fix it anyways, if they would want to make use of the parameter. But under the latest circumstances (and with the assumptions made in the Semantic Model) it makes more sense to explicitly create this restriction.

File: Sources/Apodini/Semantic Model Builder/InterfaceExporter/EnrichedContent.swift [hunk 1]
```diff
@@ -33,48 +32,75 @@ public struct EnrichedContent: Encodable {
...
     ) -> Formatter.Result {
         let context = ResolveContext(content: response.wrappedValue, parameters: parameters)
 
         var result = initialValue
 
-        if let specifiedOperation = operation {
-            endpoint.selfRelationship(for: specifiedOperation)?.formatRelationship(with: formatter, result: &result, context: context)
-        } else {
-            result = endpoint.selfRelationships().formatRelationships(into: result, with: formatter, context: context)
+        for operation in Operation.allCases.sorted(by: sortKeyPath) {
```
Comment (nityanandaz) [inline Sources/Apodini/Semantic Model Builder/InterfaceExporter/EnrichedContent.swift:79]: You need to move `Collection+SortedByKeyPath.swift` back into the `Apodini` target 🙂

Comment (bauer-andreas) [reply Sources/Apodini/Semantic Model Builder/InterfaceExporter/EnrichedContent.swift:79]: Yeah saw it, but couldn’t yet commit it.
Is it okay to make that public? We would need to as it is used in the ProtobufferExporter.

**Pull request merged**